dosma/core/med_volume.py

```diff
@@ -637,7 +643,50 @@ def to_torch(
             tensor = tensor.contiguous()
         return tensor
 
-    def headers(self, flatten=False):
+    def to_zarr(
```

Comment:
There are two design patterns (DPs) in dosma for converting MedicalVolumes to other data types:
1. I/O DP: This design pattern is used when a MedicalVolume needs to be loaded from or written to disk. Examples following this DP extend `Reader` and `Writer`, which handle all of the logic for reading/writing. Examples include DICOM and NIfTI, where there are explicit file types storing the data. 
2. Memory DP: This DP is used when a MedicalVolume is converted into a different in-memory type. This conversion is done as instance/class methods in `MedicalVolume`. Examples include torch.Tensor. There is no file writing done here.

I would argue that Zarr should follow the I/O DP because there is an explicit file that is written to or read from. In this case, there wouldn't be `to_zarr` and `from_zarr` methods in `MedicalVolume` just `ZarrReader` and `ZarrWriter`. Users would load these with:

```python
volume = dm.load("/path/to/zarr", ...)
```

Reply:
I would argue that Zarr would benefit from both approaches here or Memory DP *only*. Zarr defaults to `KVStore` which is an in-memory store (or `MemoryStore` that stores a hierarchy of `KVStore` objects), which would make it a better fit for the Memory DP. Only if you explicitly specify a `store` that supports persistence, it will write/sync to disk. 

DICOM and Nifti are specific file storage formats, whereas Zarr can be stored in a variety of ways. For Zarr, the user needs to specify the type of `store` (e.g. `DirectoryStore` vs. `ZipStore`) **before** they're able to read/write. `dm.load("/path/to/zarr")` would currently not work with a `DirectoryStore`, because `get_image_data_format` defaults to DICOM whenever it finds a directory instead of a file. 

Maybe we should remove `ZarrReader` and `ZarrWriter` given that they provide no additional functionality and there's no default "storage" for Zarr nor an associated file extension?

Reply:
Out of curiosity, what are the reasons to have `to_nib` and `from_nib` on `MedicalVolume` given the explanation of the DPs?

Reply:
got it, this all makes sense. id agree, we should get rid of the I/O DP and keep the memory DP

Reply:
to_nib and from_nib give you in-memory nibabel objects. it does not save to a nifti file